exporters/otlp-http/metrics/src/main/java/io/opentelemetry/exporter/otlp/http/metric/OtlpHttpMetricExporter.java

```diff
@@ -0,0 +1,230 @@
+import com.google.rpc.Status;
+import io.opentelemetry.api.metrics.BoundLongCounter;
+import io.opentelemetry.api.metrics.GlobalMeterProvider;
+import io.opentelemetry.api.metrics.LongCounter;
+import io.opentelemetry.api.metrics.Meter;
+import io.opentelemetry.api.metrics.common.Labels;
+import io.opentelemetry.exporter.otlp.internal.MetricAdapter;
+import io.opentelemetry.proto.collector.metrics.v1.ExportMetricsServiceRequest;
+import io.opentelemetry.sdk.common.CompletableResultCode;
+import io.opentelemetry.sdk.internal.ThrottlingLogger;
+import io.opentelemetry.sdk.metrics.data.MetricData;
+import io.opentelemetry.sdk.metrics.export.MetricExporter;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+import okhttp3.Call;
+import okhttp3.Callback;
+import okhttp3.Headers;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+import okhttp3.ResponseBody;
+import okio.BufferedSink;
+import okio.GzipSink;
+import okio.Okio;
+
+/** Exports metrics using OTLP via HTTP, using OpenTelemetry's protobuf model. */
+@ThreadSafe
+public final class OtlpHttpMetricExporter implements MetricExporter {
+
+  private static final String EXPORTER_NAME = OtlpHttpMetricExporter.class.getSimpleName();
+  private static final Labels EXPORTER_NAME_LABELS = Labels.of("exporter", EXPORTER_NAME);
+  private static final Labels EXPORT_SUCCESS_LABELS =
+      Labels.of("exporter", EXPORTER_NAME, "success", "true");
+  private static final Labels EXPORT_FAILURE_LABELS =
+      Labels.of("exporter", EXPORTER_NAME, "success", "false");
+
+  private static final MediaType PROTOBUF_MEDIA_TYPE = MediaType.parse("application/x-protobuf");
+
+  private final ThrottlingLogger logger =
+      new ThrottlingLogger(Logger.getLogger(OtlpHttpMetricExporter.class.getName()));
+
+  private final BoundLongCounter metricsSeen;
+  private final BoundLongCounter metricsExportedSuccess;
+  private final BoundLongCounter metricsExportedFailure;
+
+  private final OkHttpClient client;
+  private final String endpoint;
+  @Nullable private final Headers headers;
+  private final boolean isCompressionEnabled;
+
+  OtlpHttpMetricExporter(
+      OkHttpClient client,
+      String endpoint,
+      @Nullable Headers headers,
+      boolean isCompressionEnabled) {
+    Meter meter = GlobalMeterProvider.getMeter("io.opentelemetry.exporters.otlp-http");
+    this.metricsSeen =
+        meter.longCounterBuilder("metricsSeenByExporter").build().bind(EXPORTER_NAME_LABELS);
+    LongCounter metricsExportedCounter =
+        meter.longCounterBuilder("metricsExportedByExporter").build();
+    this.metricsExportedSuccess = metricsExportedCounter.bind(EXPORT_SUCCESS_LABELS);
+    this.metricsExportedFailure = metricsExportedCounter.bind(EXPORT_FAILURE_LABELS);
```

Comment:
we don't record metrics for the grpc metric exporter, so I wouldn't do it here. I think it's redundant since you'd see the metrics in wherever you were exporting the metrics to. :)

Reply:
Yeah I noticed that about the grpc metric exporter. Figured it was an "add later" omission like TLS support.

On one hand, with these types of metrics you can capture information about failed exports in a way that can't be observed by the server you're exporting to. Although, if the failed exports aren't intermittent, then the metrics are unlikely to make it to a sink where you can observe them.

Reply:
If we want to add metrics to our metrics export, let's do it all at one time, rather than just here. Sound ok?